cmd/codegen/main.go

```diff
@@ -112,18 +116,43 @@ func ClientGen(cmd *cobra.Command, args []string) error {
 		cfg.IntrospectionJSON = string(introspectionJSON)
 	}
 
-	if dependenciesJSONFilePath != "" {
-		dependenciesJSON, err := os.ReadFile(dependenciesJSONFilePath)
+	if moduleSourceID != "" {
+		dag, err := dagger.Connect(ctx)
```

Comment:
Is this two separate connections to the engine, one in `codegen.go` as well? That's a bit confusing if so, we should only create one connection.

Reply:
Yeah I could pass the `dag` as parameter to the `generate` interface function, so the caller is reponsible for opening/closing the connection.

Reply:
Fixed, I added a field `Dag` in the config because depending on the context we cannot connect to the engine.
So it checks which flag are set and will connect to the engine if it makes sense to